Dockerfile

```diff
@@ -17,12 +12,18 @@ FROM gostable AS go-linux
 # These will be used at current arch to yield execute the cross compilations.
 FROM go-${TARGETOS} AS build-base
 
-RUN apk add clang lld pkgconfig
+RUN apk add clang lld
 
 COPY --from=xx / /
 
-# build-go-mod can still be cached at build platform architecture.
-FROM build-base as build-go-mod
+# build can still be cached at build platform architecture.
+FROM build-base as build
+
+ARG TARGETPLATFORM
+
+# Some dependencies have to installed
+# for the target platform: https://github.com/tonistiigi/xx#go--cgo
+RUN xx-apk add musl-dev gcc
```

Comment:
Ref recommends explicit declaration of `CGO_ENABLED`.

In addition: might be worth taking note of a brief "why" for the CGO dependency.